mh/pat.py

```diff
@@ -1131,10 +1133,13 @@ def sendAnsBinaryHead(self, binary_type, seq):
          - Spanish: 0x41, 0x3d, 0x41, 0x3e, 0x3f, 0x40
         """
         binary = PAT_BINARIES[binary_type]
-        content = binary["content"]
+        version = binary["version"]
+        if callable(version):
+            version = version(self.server.binary_loader)
+        content = get_pat_binary_from_version(binary_type, version)
         if callable(content):
-            content = content()
-        data = struct.pack(">II", binary["version"], len(content))
+            content = content(self.server.binary_loader)
+        data = struct.pack(">II", version, len(content))
```

Comment:
Can't this logic/part be offloaded to the `binary_loader`?

Reply:
Not all binaries currently come from the `binary_loader`, only the ones that have the potential to change from moment to moment. In other words yes, it could be moved (and probably should, once we have things like a rotating trading post), but since this PR is specifically about the event quests, I'll leave that to a future update.